From 6f4d8d43166e37052a39a87e36be6cb6604bfe88 Mon Sep 17 00:00:00 2001
From: Steev Klimaszewski <threeway@gmail.com>
Date: Fri, 23 Jan 2026 18:40:10 -0600
Subject: [PATCH] arm64: dts: qcom: sc8280xp-x13s: add EC

Describe ThinkPad Embedded Controller in the X13s device tree,
which adds LED and special key support.

Signed-off-by: Steev Klimaszewski <threeway@gmail.com>
---
 .../qcom/sc8280xp-lenovo-thinkpad-x13s.dts    | 20 +++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts b/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts
index 18fcad10438cf6..ad86c8b97f1893 100644
--- a/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts
+++ b/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts
@@ -811,6 +811,20 @@
 	};
 };
 
+&i2c8 {
+	clock-frequency = <400000>;
+	status = "okay";
+
+	embedded-controller@28 {
+		compatible = "lenovo,thinkpad-x13s-ec";
+		reg = <0x28>;
+		interrupts-extended = <&tlmm 176 IRQ_TYPE_LEVEL_LOW>;
+		pinctrl-0 = <&ec_int_n_default>;
+		pinctrl-names = "default";
+		wakeup-source;
+	};
+};
+
 &i2c11 {
 	clock-frequency = <400000>;
 
@@ -1573,6 +1587,12 @@
 		};
 	};
 
+	ec_int_n_default: ec-int-n-state {
+		pins = "gpio176";
+		functions = "gpio";
+		bias-disable;
+	};
+
 	edp_reg_en: edp-reg-en-state {
 		pins = "gpio25";
 		function = "gpio";
